examples/dcmip_3_1_meanflow_quads.py

```diff
@@ -0,0 +1,190 @@
+from gusto import *
+from firedrake import CubedSphereMesh, ExtrudedMesh, Expression, \
+    FunctionSpace, Function, SpatialCoordinate, as_vector
+from firedrake import exp, acos, cos, sin, pi
+import sys
+
+
+nlayers = 10           # Number of vertical layers
+refinements = 3        # Number of horiz. refinements
+
+dt = 100.0             # Time-step size (s)
+
+if '--running-tests' in sys.argv:
+    tmax = dt
+else:
+    tmax = 3600.0
+
+
+parameters = CompressibleParameters()
+a_ref = 6.37122e6               # Radius of the Earth (m)
+X = 125.0                       # Reduced-size Earth reduction factor
+a = a_ref/X                     # Scaled radius of planet (m)
+g = parameters.g                # Acceleration due to gravity (m/s^2)
+N = parameters.N                # Brunt-Vaisala frequency (1/s)
+p_0 = parameters.p_0            # Reference pressure (Pa, not hPa)
+c_p = parameters.cp             # SHC of dry air at constant pressure (J/kg/K)
+R_d = parameters.R_d            # Gas constant for dry air (J/kg/K)
+kappa = parameters.kappa        # R_d/c_p
+T_eq = 300.0                    # Isothermal atmospheric temperature (K)
+p_eq = 1000.0 * 100.0           # Reference surface pressure at the equator
+u_max = 20.0                    # Maximum amplitude of the zonal wind (m/s)
+d = 5000.0                      # Width parameter for Theta'
+lamda_c = 2.0*pi/3.0            # Longitudinal centerpoint of Theta'
+phi_c = 0.0                     # Latitudinal centerpoint of Theta' (equator)
+deltaTheta = 1.0                # Maximum amplitude of Theta' (K)
+L_z = 20000.0                   # Vertical wave length of the Theta' perturb.
+
+# Cubed-sphere horizontal mesh
+m = CubedSphereMesh(radius=a,
+                    refinement_level=refinements,
+                    degree=2)
+
+# Build volume mesh
+z_top = 1.0e4                  # Height position of the model top
+mesh = ExtrudedMesh(m, layers=nlayers,
+                    layer_height=z_top/nlayers,
+                    extrusion_type="radial")
+
+# Create polar coordinates:
+# Since we use a CG1 field, this is constant on layers
+W_Q1 = FunctionSpace(mesh, "CG", 1)
+z_expr = Expression("sqrt(x[0]*x[0] + x[1]*x[1] + x[2]*x[2]) - a", a=a)
+z = Function(W_Q1).interpolate(z_expr)
+lat_expr = Expression("asin(x[2]/sqrt(x[0]*x[0] + x[1]*x[1] + x[2]*x[2]))")
```

Comment:
Does this need to be a c expression or can you write it with Firedrake functions?

Reply:
No, you are right...thanks for holding me accountable :P I will update.